### ISSUE TITLE
How can I include macro variables?

### ISSUE THREAD

**Issue by lilli0204** [lilli0204/including-macro-variables#1]
Hello, 

I'm currently trying to 
add macro variables in my model. Do you think the code is correct? 

# 🔹 1️⃣ 
selected_financial_features = final_filter_features  # z. B. ['EKQ', 'FKQ', 'ROA', ...]
macro_features = ['Inflationsrate_Lag', 'BIP_Wachstum_Lag', 'Arbeitslosenquote_Lag']
target = 'Insolvency_Flag'

# 🔹 2️⃣ 
X_base = df_firm[selected_financial_features + macro_features].copy()
y_all = df_firm[target]

# 🔹 3️⃣ 
jahr_dummies = pd.get_dummies(df_firm['FZ_JAHR'], prefix='Jahr', drop_first=True)
X_all = pd.concat([X_base, jahr_dummies], axis=1)

# 🔹 5️⃣ Train-Test-Split
X_train_extended, X_test_extended, y_train, y_test = train_test_split(
    X_all, y_all, test_size=0.3, random_state=42, stratify=y_all
)

# 🔹 6️⃣ Entscheidungsbaum-Instanz
dt_extended = DecisionTreeClassifier(random_state=42)

# 🔹 7️⃣ Hyperparameter-Raster für GridSearchCV
param_grid = {
    'max_depth': [3, 5, 10, 15, None],
    'min_samples_split': [2, 5, 10],
    'min_samples_leaf': [1, 2, 4]
}

# 🔍 8️⃣ GridSearchCV ausführen
grid_search_extended = GridSearchCV(
    estimator=dt_extended,
    param_grid=param_grid,
    cv=5,
    scoring='roc_auc',
    n_jobs=-1,
    verbose=1
)
grid_search_extended.fit(X_train_extended, y_train)

# 🔍 9️⃣ Bestes Modell extrahieren
best_dt_extended = grid_search_extended.best_estimator_

# 🔹 🔟 Vorhersagen auf Testdaten
y_pred_extended = best_dt_extended.predict(X_test_extended)
y_proba_extended = best_dt_extended.predict_proba(X_test_extended)[:, 1]

# 🔹 11️⃣ Metriken ausgeben
def print_metrics(y_true, y_pred, y_proba, title):
    print(f"\n🔹 {title}")
    print(f"AUC:         {roc_auc_score(y_true, y_proba):.4f}")
    print(f"Genauigkeit: {accuracy_score(y_true, y_pred):.4f}")
    print(f"Precision:   {precision_score(y_true, y_pred):.4f}")
    print(f"Recall:      {recall_score(y_true, y_pred):.4f}")
    print(f"F1-Score:    {f1_score(y_true, y_pred):.4f}")
    print(classification_report(y_true, y_pred))

print_metrics(y_test, y_pred_extended, y_proba_extended, "Entscheidungsbaum – finanziell + Makro + Jahr")
